perf_test/graph/KokkosGraph_color_d2.cpp

```diff
@@ -539,8 +537,8 @@ int main (int argc, char ** argv)
   std::cout << "Sizeof(kk_lno_t) : " << sizeof(kk_lno_t) << std::endl 
             << "Sizeof(size_type): " << sizeof(kk_size_type) << std::endl;
 
-#if defined( KOKKOS_HAVE_OPENMP )
 
```

Comment:
Don't do this explicit OpenMP initialization. We are deprecating this. Call some form of Kokkos::initialize() instead.

Reply:
@william76 @crtrott 

Is there an example of running OMP execution space when Cuda is enabled?

Reply:
@crtrott
Happy to change it, but need to know what to change it to... 

@mndevec if we have to change it here, we'll also need to change the distance-1 coloring code too since this shares a lot with that driver, right?

Reply:
@william76  : yes, this needs to be changed for all perf_test. 
If you can fix it in this, I would follow your example to fix it for the rest of the tests.

Reply:
@mndevec ok... so being new to `Kokkos`, is there a place to look to find what variant of `Kokkos::initialize()` that I should go for?  I'm thinking there's probably one that goes something like:  `Kokkos::initialize(argc, argv)`  ?

Is there an API available that does the same thing as `KOKKOS_HAVE_OPENMP` ?

Reply:
You don't need KOKKOS_HAVE_OPENMP (which btw. is now KOKKOS_ENABLE_OPENMP).
For initialization syntax look here:
https://github.com/kokkos/kokkos/wiki/Initialization

Reply:
@mndevec
@crtrott

I'm still confused on this... the main() in [KokkosGraph_color_d2.cpp][1] has several sections that look like:

```
#if defined( KOKKOS_HAVE_OPENMP )
  if (params.use_openmp)
  {
    Kokkos::OpenMP::initialize( params.use_openmp );
    Kokkos::OpenMP::print_configuration(std::cout);
#ifdef KOKKOSKERNELS_MULTI_MEM
    KokkosKernels::Experiment::run_multi_mem_experiment
        <kk_size_type, kk_lno_t, Kokkos::OpenMP, Kokkos::OpenMP::memory_space, Kokkos::HostSpace>(params);
#else
    KokkosKernels::Experiment::run_multi_mem_experiment
        <kk_size_type, kk_lno_t, Kokkos::OpenMP, Kokkos::OpenMP::memory_space, Kokkos::OpenMP::memory_space>(params);
#endif
    Kokkos::OpenMP::finalize();
  }
#endif
```

so, I can put in a `Kokkos::initialize(argc, argv)` and `Kokkos::finalize()` at the end of the function but I'm not sure about the modifications needed to the driver.

I imagine `Kokkos::OpenMP::initialize()` can go away, but there's still stuff like `Kokkos::OpenMP::print_configuration()`, `Kokkos::OpenMP::memory_space`, and the OpenMP parameters that are getting passed around inside the `#if defined(KOKKOS_HAVE_OPENMP)` guard.  What do I do with those? Will they need to change? Are there analogs for them if they do?  Sadly, Kokkos doxygen pages aren't really helpful and I didn't write this file so it's not completely clear what the modifications are.  Are there any examples in Kokkos that show what the pattern should look like?

If the delta is small to fix this issue, then I think we can put it in with this PR... otherwise, I think this is outside of the scope of the purpose of this particular PR, which is to fix the CUDA bug in the D2 coloring kernel, and we should merge this PR if it's fixing the bug in the D2 coloring algorithm.  I think we should put in a separate issue ticket for the restructuring of this code and address the specific changes there.

[1]: https://github.com/kokkos/kokkos-kernels/blob/6e8e97a977564673bdadc15085d27a6a56b98114/perf_test/graph/KokkosGraph_color_d2.cpp#L542

Reply:
@crtrott your requests are beyond the scope of this Pull Request.  I have created an issue, #200, to track the change request and broaden the scope of that to all the tests in `perf_test/graph`.  

Assuming the updates in this PR sufficiently address the bugs in the D2 Graph Coloring implementation, do you mind merging this PR in and we can work on the other stuff off of #200?